src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -427,6 +440,20 @@ private String getMapping() throws IOException {
             return Resources.toString(url, Charsets.UTF_8);
         }
 
+        @Override
+        public boolean isModelBlockedForDelete(String modelId) {
+            // Check if the objects are not null and throw a customized NullPointerException
+            Objects.requireNonNull(clusterService.state(), "Cluster state must not be null");
+            Objects.requireNonNull(clusterService.state().metadata(), "Cluster metadata must not be null");
+            ModelGraveyard modelGraveyard = clusterService.state().metadata().custom(ModelGraveyard.TYPE);
+
+            if (isNull(modelGraveyard)) {
```

Comment:
So in the case where a model is being trained on a previous version node, this will return here - is that correct?

Reply:
Do you mean the validation against modelGraveyard for train model request ?

Reply:
In the case during a mixed cluster, when training is running on an old node and a request comes to the new node to delete the model, this method would return false. But I dont think we can do anything to fix this. So Ill resolve.